### ISSUE TITLE
azuremonitorexporter assign wrong parentId to span events

### ISSUE THREAD

**Issue by Obi-Dann** [open-telemetry/opentelemetry-collector-contrib#27233]
### Component(s)

exporter/azuremonitor

## Description
`azuremonitorexporter` seems to assign wrong parentId to span events. As the results, span events have the same parentId as the span, but should have spanId as the parentId IIUC.

![image](https://github.com/open-telemetry/opentelemetry-collector-contrib/assets/12251341/43be8622-d128-4a52-8a49-7d8d0681a49a)
![image](https://github.com/open-telemetry/opentelemetry-collector-contrib/assets/12251341/b2317089-6588-45cf-bc81-5b19bbbd5d0b)

It looks like the issue is [here](https://github.com/pcwiese/opentelemetry-collector-contrib/blob/3a6f54773c1c494bb95e31307c2faefa6bd38403/exporter/azuremonitorexporter/trace_to_envelope.go#L134)
`newEnvelope` implementation should be different for span events, setting parentId to the spanId instead of [always setting](https://github.com/pcwiese/opentelemetry-collector-contrib/blob/3a6f54773c1c494bb95e31307c2faefa6bd38403/exporter/azuremonitorexporter/trace_to_envelope.go#L179C2-L179C2) parentId to the span's parent id

## Steps to Reproduce
Configure a collector with azuremonitor as an exported, set `spaneventsenabled: true` and ensure to send spans with events to the collector.

## Expected Result
Span events should have the spanId set as their parentId

## Actual Result
Span events have the span's parentId set as their parentId

### Collector version

0.85.0

## Environment
OS: macOS 13.6

### OpenTelemetry Collector configuration

```yaml
receivers:
  otlp:
    protocols:
      http:

exporters:
  logging:
  azuremonitor:
    instrumentation_key: ""
    spaneventsenabled: true

processors:
  batch:
    timeout: 10s
    send_batch_size: 1000

service:
  pipelines:
    traces:
      receivers: [otlp]
      processors: []
      exporters: [logging, azuremonitor]
    logs:
      receivers: [otlp]
      processors: []
      exporters: [logging, azuremonitor]
```

**Comment by crobert-1** [comment]
The semantic conventions are a bit unclear here as there's no default attribute in a span event for parentID. I think your interpretation is fair in that it makes sense for the events of a span to be considered the children of the span itself. Wording it another way, the span can be considered a parent of all of its own events.

@pcwiese may have different thoughts as code owner though, and this may also break some people's existing assumptions and queries in the backend.

Relevant documentation: [Semantic conventions](https://opentelemetry.io/docs/specs/otel/trace/api/#add-events)

**Comment by alaendle** [comment]
We also encountered this issue while forwarding Zipkin events to Azure Monitor.
This current implementation just creates wrong impressions - the traces are one level-up in the tree; so one could expect that the trace was generated by the producer - even if the consumer produces the trace.
See the following AppInsights screenshot (I've added the red circles/arrows to illustrate the problem):
![grafik](https://github.com/open-telemetry/opentelemetry-collector-contrib/assets/969523/c45db93b-00fb-4589-b891-affc53eedcd0)
Please let me know if I could provide further information.

**Comment by alaendle** [comment]
This is solved as soon as #30659 is merged - so if anyone with rights to merge this PR reads this, please make progress at this point 📯 .

**Comment by crobert-1** [comment]
Resolved by https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/30659
